scripts/SANS/sans/algorithm_detail/strip_end_nans_and_infs.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import (absolute_import, division, print_function)
+from math import (isinf, isnan)
+from sans.common.constants import EMPTY_NAME
+from sans.common.general_functions import create_child_algorithm
+
+
+def strip_end_nans(workspace, parent_alg=None):
+    """
+    This function removes the INFs and NANs from the start and end of a 1D workspace.
+
+    :param workspace: The workspace which is about to be
+    :param parent_alg: a handle to the parent algorithm
+    :return: A trimmed NAN- and INF-trimmed workspace
+    """
+    # If the workspace is larger than 1D, then there is nothing we can do
+    if workspace.getNumberHistograms() > 1:
+        return workspace
+    data = workspace.readY(0)
+    # Find the index at which the first legal value appears
+
+    start_index = next((index for index in range(0, len(data)) if is_valid_data(data[index])), None)
+    end_index = next((index for index in reversed(list(range(0, len(data)))) if is_valid_data(data[index])), None)
```

Comment:
this is too complicated `reversed(list(range(0, len(data))))` should be `range(len(data)-1,-1,-1)`